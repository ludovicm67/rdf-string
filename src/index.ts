export { sparql, SparqlTemplateResult, SparqlValue } from './lib/sparql'
