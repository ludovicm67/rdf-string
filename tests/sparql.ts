import { sparql } from '../src/index'
import { namedNode } from '@rdfjs/data-model'
import { prefixes } from '@zazuko/rdf-vocabularies'
import namespace from '@rdfjs/namespace'

const schema = namespace(prefixes.schema)

describe('sparql', () => {
  it('serializes named node', () => {
    // given
    const type = namedNode('http://example.com/type')

    // when
    const query = sparql`SELECT * WHERE { ?person a ${type} }`

    // then
    expect(query.toString()).toMatchSnapshot()
  })

  it('serializes string literal', () => {
    // given
    const name = 'John Doe'

    // when
    const query = sparql`SELECT * WHERE { ?person <http://schema.org/name> ${name} }`

    // then
    expect(query.toString()).toMatchSnapshot()
  })

  it('extracts known prefixes', () => {
    // when
    const query = sparql`SELECT * WHERE { ?person ${schema.name} "Tomasz" }`

    // then
    expect(query.toString()).toMatchSnapshot()
  })
})
