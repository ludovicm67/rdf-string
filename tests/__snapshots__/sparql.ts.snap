// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sparql extracts known prefixes 1`] = `
"PREFIX schema: <http://schema.org/>

SELECT * WHERE { ?person schema:name \\"Tomasz\\" }"
`;

exports[`sparql serializes named node 1`] = `"SELECT * WHERE { ?person a <http://example.com/type> }"`;

exports[`sparql serializes string literal 1`] = `"SELECT * WHERE { ?person <http://schema.org/name> John Doe }"`;
